src/languages/php.js

```diff
@@ -18,18 +19,50 @@ export default function(hljs) {
       { begin: /\?>/ } // end php tag
     ]
   };
+  var SUBST ={
+    className: 'subst',
+    variants: [
+      {begin: /\$\w+/},
+      {begin: /\{\$/, end: /\}/}
+    ]
+  };
+  var SINGLE_QUOTED = hljs.inherit(
+    hljs.APOS_STRING_MODE,
+    {
+      illegal: null,
+      variants: [{
+        begin: 'b\'', end: '\''
+      }]
+    }
+  );
+  var DOUBLE_QUOTED = hljs.inherit(
+    hljs.QUOTE_STRING_MODE,
+    {
+      illegal: null,
+      variants: [{
+        begin: 'b"', end: '"'
+      }],
+      contains: [hljs.BACKSLASH_ESCAPE, SUBST]
+    }
+  );
```

Comment:
Pretty sure this is wrong and *not* the equivalent of what we were doing before, it's also MUCH harder to read. (it's unclear that there are two branches).  Pretty sure variants needs to spell out ALL the variants. You're also making every variant null, which is also incorrect, etc...

Variants are a complete list not "in addition to"....

I'd make another attempt here or just revert this to what we had before.

Reply:
Maybe I misunderstood thinking the variants of the main will result in the main + expansion for each variant, isn't it ? It was working as expected testing it, so how ? I wanted to reflect the fact that each string (single and double) has its own "binary" equivalent for PHP6 compatibility...

Reply:
> Maybe I misunderstood thinking the variants of the main will result in the main + expansion for each variant, isn't it?

No.  Variants are exclusive.  

> I wanted to reflect the fact that each string (single and double) has its own "binary" equivalent for PHP6 compatibility...

I think we can just do this with comments the way this was prior. Otherwise you're repeating a lot of the string bits over and over for no real big win...

Reply:
Ok so I noticed when retesting my code that binary strings were broken so I fixed it. I seems the variants where not expanded recursively, a bug maybe ? (Prior the substitution was not used in binary strings, surely not a big win because it surely affects nobody as it is a backward compatiblity residue but seems clearer...)